src/main/java/com/github/ragurney/spotless/actions/ReformatSelectedFilesAction.java

```diff
@@ -0,0 +1,58 @@
+package com.github.ragurney.spotless.actions;
+
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.changes.ui.ChangesListView;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class ReformatSelectedFilesAction extends AnAction {
+
+	@Override
+	public void actionPerformed(@NotNull AnActionEvent event) {
+		DataContext dataContext = event.getDataContext();
+
+		final Project project = CommonDataKeys.PROJECT.getData(dataContext);
+		if (project == null) {
+			return;
+		}
+		PsiManager manager = PsiManager.getInstance(project);
+		Stream
+				.concat(getExactlySelectedFiles(manager, event).stream(), selectedChangesetFiles(manager, event).stream())
+				.forEach(file -> new ReformatCodeProcessor(file, false).run());
```

Comment:
Unfortunately, this does not work since we can only run one gradle process at a time on a project. If you try to run this with multiple files, you will see an error such as:
![image](https://user-images.githubusercontent.com/15261525/216356877-be32114f-e722-40c7-adb4-8faef38ac657.png).

This is the same issue that #33 ran into. In order for this to be fixed and for me to accept this change, we'll need to implement or push for some progress on https://github.com/diffplug/spotless/issues/791. 

Another alternative is to switch over to using IntelliJ's External Formatter API which I attempt to do in [this branch](https://github.com/ragurney/spotless-intellij-gradle/tree/RefactorActionToExternalFormatterAPI), but this would drastically change the UX of the plugin, and I was running into some multithreading issues :(